<launch>
    
    <!--- Sim Time -->
    <!-- <param name="/use_sim_time" value="true" /> -->
    <!-- <param name="/use_sim_time" value="false" /> -->

    <!--- Run Rviz-->
    <node pkg="rviz" type="rviz" name="rviz" args="-d $(find lego_loam)/launch/test.rviz" />

    <!--- TF -->
    <!-- <node pkg="tf" type="static_transform_publisher" name="camera_init_to_map"  args="0 0 0 1.570795   0        1.570795 /map    /camera_init 10" /> -->
    <!-- <node pkg="tf" type="static_transform_publisher" name="base_link_to_camera" args="0 0 0 -1.570795 -1.570795 0        /camera /base_link   10" /> -->

    <!-- <node pkg="lego_loam" type="lidar_imu_sub" name="lidar_imu_sub" output="screen"/> -->

    <!--- LeGO-LOAM -->    
    <node pkg="lego_loam" type="imageProjection"    name="imageProjection"    output="screen">

       <param name="remove_range"  value="0.5" />
       <param name="groundscanind" type="int" value="7" />
       <param name="segtheta"  value="1.0472" />
       <param name="n_scan" type="int" value="16" />
       <param name="h_scan" type="int" value="1800" />
       <param name="lidarTopics" type="string" value="/velodyne_ns/velodyne_points" />
       <!-- <param name="lidarTopics" type="string" value="/velodyne_points" /> -->
       <!-- <param name="lidarTopics" type="string" value="/vio_test/velodyne" /> -->
                 

    </node>

    <node pkg="lego_loam" type="featureAssociation" name="featureAssociation" output="screen">

       <param name="imu_topic" type="string" value="/imu_ns/imu/filter" />
       <param name="n_scan" type="int" value="16" />
       <param name="h_scan" type="int" value="1800" />

    </node>

    <node pkg="lego_loam" type="mapOptmization"     name="mapOptmization"     output="screen">

       <param name="loopflag" type="bool" value="false" />
       <param name="imu_topic" type="string" value="/imu_n" />
       <param name="mapinterval" type="double" value="0.1" />
       <param name="loop_distance"  value="20" />
       <param name="map_range"  value="500.0" />
       <param name="number_keyframe"  value="50.0" />
       <param name="indoor_flag" type="bool" value="false" />

    </node>

    <node pkg="lego_loam" type="transformFusion"    name="transformFusion"    output="screen">

    </node>

</launch>
